.build/docker/ubuntu2004_test.docker

```diff
@@ -146,11 +131,26 @@ RUN /bin/bash -c "export CASS_DRIVER_NO_CYTHON=1 CASS_DRIVER_NO_EXTENSIONS=1 \
     && pip3 freeze --user"
 
 # Initialize the CCM git repo as well as this also can fail to clone
-RUN /bin/bash -c "source ${BUILD_HOME}/env3.6/bin/activate && \
-    ccm create -n 1 -v git:trunk test && ccm remove test && \
+RUN /bin/bash -c "source ${BUILD_HOME}/env3.7/bin/activate && \
     ccm create -n 1 -v git:cassandra-4.1 test && ccm remove test && \
     ccm create -n 1 -v git:cassandra-4.0 test && ccm remove test"
 
+# Initialize ccm versions.  right side of each sequence needs to be updated with new releases.
+#  this can be checked with:
+#  `curl -s https://downloads.apache.org/cassandra/ | grep -oP '(?<=href=\")[0-9]+\.[0-9]+\.[0-9]+(?=)' | sort -V | uniq -w 3`
+RUN bash -c 'source ~/env3.7/bin/activate && \
+    for i in {1..11} ; do echo $i ; ccm create --quiet -n 1 -v binary:4.0.$i test && ccm remove test ; done && \
```

Comment:
So we're intentionally trying to init 4.0.1 to 4.0.11 right? And when a 4.0.12 hits we'll need to manually add that here?

If we want to be... clever? Hacky? We could do something like:
`git tag | grep -E 'cassandra-4.0.[0-9]+$' | cut -d '.' -f 3 | sort -nr | head -n 1`

to parse out the latest version at time of docker image creation and iterate through all up to that point to init them. Alternatively, or along with that, might be nice to add a comment to indicate why we have the top end of the sequence here since it's non-obvious.

Reply:
I thought about it too, but this approach may fail the CI if the git tag is added before the actual release happens, right? That's why fetching from the archive is better, I think...